src/HamamatsuCamera.cpp

```diff
@@ -3052,78 +3051,80 @@ enum Camera::Output_Trigger_Kind Camera::getOutputTriggerKind(int channel)
 //-----------------------------------------------------------------------------
 enum Camera::Output_Trigger_Polarity Camera::getOutputTriggerPolarity(int channel)
 {
-    
     DEB_MEMBER_FUNCT();
 
-    DEB_TRACE() << "Camera::getOutputTriggerPolarity(int channel) : channel = " << channel ;
+    DEB_TRACE() << "Camera::getOutputTriggerPolarity(int channel) : channel = " << channel;
 
     enum Output_Trigger_Polarity result = Output_Trigger_Polarity::Output_Trigger_Polarity_Not_Supported;
 
     DCAMERR err;
     double polarityArraySize = 0;
     enum Output_Trigger_Polarity polarity = Camera::Output_Trigger_Polarity_Not_Supported;
 
-	// get property attribute that contains the Polarity (that may be an array)
-	DCAMPROP_ATTR	basepropattr;
-	memset(&basepropattr, 0, sizeof(basepropattr));
-	basepropattr.cbSize = sizeof(basepropattr);
-	basepropattr.iProp = DCAM_IDPROP_OUTPUTTRIGGER_POLARITY;
-	err = dcamprop_getattr(m_camera_handle, &basepropattr);
+    // get property attribute that contains the Polarity (that may be an array)
+    DCAMPROP_ATTR basepropattr;
+    memset(&basepropattr, 0, sizeof(basepropattr));
+    basepropattr.cbSize = sizeof(basepropattr);
+    basepropattr.iProp = DCAM_IDPROP_OUTPUTTRIGGER_POLARITY;
+    err = dcamprop_getattr(m_camera_handle, &basepropattr);

```

Comment:
possibility to use getPropertyData() ?

Reply:
Yes, but in this case we manage error from get Hamamatsu data for kind and polarity

Reply:
getPropertyData is not yet able to get the number of element when the requested attribute is an array. Dealing with the attribute structure directly allow us to get that number of element which is then used to ensure that the input value (channel) is not beyond the sensor capabilities.

Such behaviour may be added if required.